c2corg_api/search/search.py

```diff
@@ -0,0 +1,63 @@
+from elasticsearch_dsl.query import MultiMatch
+from elasticsearch_dsl.filter import Term
+from sqlalchemy.orm import joinedload
+
+from c2corg_api.models import DBSession
+from c2corg_api.search import create_search
+from c2corg_api.views import to_json_dict
+
+
+def search_for_type(
+        search_term, document_type, model, schema, adapt_schema, limit):
+    # search in all title* (title_en, title_fr, ...), summary* and
+    # description* fields. "boost" title fields and summary fields.
+    search_query = MultiMatch(
+        query=search_term,
+        fields=['title*^3', 'summary*^2', 'description*']
+    )
+
+    # filter on the document_type
+    type_query = Term(doc_type=document_type)
+
+    search = create_search().\
+        query(search_query).\
+        filter(type_query).\
+        fields([]).\
+        extra(from_=0, size=limit)
+
+    # only request the document ids from ES
+    response = search.execute()
+    document_ids = [int(doc.meta.id) for doc in response]
+
+    # then load the documents for the returned ids
+    documents = get_documents(document_ids, model)
+
+    count = len(documents)
+    total = response.hits.total
+
+    return {
+        'count': count,
+        'total': total,
+        'documents': [
+            to_json_dict(doc, adapt_schema(schema, doc)) for doc in documents
+        ]
+    }
+
+
+def get_documents(document_ids, model):
+    """Load the documents for the given ids.
+    The documents are returned in the same order as the ids. If a document
+    for a given id does not exist, the document is skipped.
+    """
+    if not document_ids:
+        return []
+
+    documents = DBSession.\
+        query(model).\
```

Comment:
Yes, this would also be relevant for the listings. I was thinking about creating a separate model for the listings which only contains the fields needed for the listings. I will create an issue for it.